### PR TITLE
feat(github): convert `platformCommit` to enum

### LINKED ISSUE #29115: [GitHub] enable `platformCommit` when we detect an app token
### Describe the proposed change(s).

Change `platformCommit` to enum like with values `enabled`, `auto` and `disabled`.
Default to `auto`.
Set to `enabled` when it's auto and we detect an github app token.
i think they start with `ghs_`.

https://github.com/renovatebot/renovate/blob/66de0465e944f55669d5822cea6a2d459a6e5ed6/lib/util/check-token.ts#L67-L69

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes
- Convert `platformCommit` to enum
- Add validation tests
- Add migration both from repoConfig and from env
- Set to `enabled` if value is `auto` and platform is githubApp

## Context
Closes: https://github.com/renovatebot/renovate/issues/29115

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by rarkins** [comment]
@RahulGautamSingh I think you might want to start over by branching off v38 to avoid conflicts

**Comment by RahulGautamSingh** [comment]
@viceice What updates should be mentioned in the docs?

**Comment by RahulGautamSingh** [comment]
The failing tests is a v38 issue

File: lib/config/migrations/custom/platform-commit-migration.spec.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { PlatformCommitMigration } from './platform-commit-migration';
+
+describe('config/migrations/custom/platform-commit-migration', () => {
+  it('should migrate platformCommit=true to platformCommit=enabled', () => {
+    expect(PlatformCommitMigration).toMigrate(
+      {
+        platformCommit: true as any,
```
Comment (viceice) [inline lib/config/migrations/custom/platform-commit-migration.spec.ts:7]: use ts-expect-error instead

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -13319,4 +13319,4 @@ snapshots:
 
   zod@3.23.8: {}
 
-  zwitch@1.0.5: {}
\ No newline at end of file
+  zwitch@1.0.5: {}
```
Comment (viceice) [inline pnpm-lock.yaml:13322]: revert this file

Comment (RahulGautamSingh) [inline pnpm-lock.yaml:13322]:
```suggestion
  zwitch@1.0.5: {}
```

**Review by viceice — APPROVED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**